core/dbt/task/freshness.py

```diff
@@ -190,6 +190,15 @@ def result_path(self):
             return os.path.realpath(self.args.output)
         else:
             return os.path.join(self.config.target_path, RESULT_FILE_NAME)
+    
+    def archive_path(self):
```

Comment:
Not a fan of this behavior, and it won't be needed if we use `target_path` + `state_path` (instead of `state_path` + `state_path/archive_sources`)

Reply:
Love it - `archive_sources` didn't feel great :). Noting a question from above about how a user should configure `target_path` + `state_path` to accomplish this.